src/js/model/number/float.js

```diff
@@ -0,0 +1,99 @@
+var AMPParse = AMPParse || {};
+
+
+AMPParse.buildFloatNumber = function (hexadecimal, precision)
+{
+    // Defensive programming
+    if (typeof hexadecimal === "undefined" || typeof hexadecimal !== "string")
+    {
+        throw new ReferenceError("Provided hexadecimal parameter does not exist or is not a string");
+    }
+
+    var nibbles = 0;
+
+    if (precision === "32") {
+        nibbles = 8; // 32 bits
+    } else if (precision === "64") {
+        nibbles = 16; // 64 bits
+    } else {
+        throw new ReferenceError("Provided precision is invalid: " + precision);
+    }
+
+    if (hexadecimal.length < nibbles)
+    {
+        throw new RangeError("The provided hex is too short to contain a " + precision + ".");
+    }
+
+    var binary = "";
+    for (var i = 0; i < nibbles; i=i+2)
+    {
+        var nextChars = hexadecimal.substr(i, 2);
+        var rawValue = parseInt(nextChars, 16);
+        binary += pad(rawValue.toString(2));
+    }
+
+    // sign bit is always first regardless of size
+    var sign =  binary.charAt(0);
+
+    var exponent;
+    var mantissa;
+    if (nibbles === 8) {
+        exponent = binary.substr(1, 8);
```

Comment:
Substring's second bound is always exclusive. The correct substring methods would be `substring(1,9)` and then `substring(9)`.